spec/acceptance/users_spec.rb

```diff
@@ -0,0 +1,100 @@
+require "rspec_api_documentation/dsl"
+
+RSpec.resource "Users" do
+  let(:authenticated_user) do
+    create(:confirmed_user)
+  end
+
+  let(:auth_token) do
+    Knock::AuthToken.new(payload: { sub: authenticated_user.id }).token
+  end
+
+  shared_context "user parameters" do
+    parameter "first_name", <<-DESC, scope: :user, required: true
+      The first name of the user.
+    DESC
+    parameter "email", <<-DESC, scope: :user, required: true
+      The email of the user.
+    DESC
+    parameter "last_name", <<-DESC, scope: :user, required: true
+      The last name of the user.
+    DESC
+    parameter "password", <<-DESC, scope: :user, required: true
+      The password of the user.
+    DESC
+    parameter "city", <<-DESC, scope: :user, required: true
+      The city of the user.
+    DESC
+    parameter "state", <<-DESC, scope: :user, required: true
+      The state of the user.
+    DESC
+    parameter "country", <<-DESC, scope: :user, required: true
+      The country of the user.
+    DESC
+  end
+
+  post "/users" do
+    include_context "user parameters"
+
+    let(:first_name) { "Nate" }
+    let(:email)      { "admin@test.com" }
+    let(:last_name)  { "Pauzenga" }
+    let(:city)       { "Annapolis" }
+    let(:state)      { "MD" }
+    let(:country)    { "USA" }
+    let(:password)   { "helloworld" }
+
+    example_request "POST /users" do
+      expect(status).to eq 201
+      json_response = JSON.parse(response_body)
+      expect(json_response["jwt"]).to include("payload")
+    end
+  end
+
+  patch "/users/:id" do
```

Comment:
I think we talked about this one but just in case here's how I believe it should work: for the most part users would not need to update some other user's profile/account. This might be functionality that an admin of the system might want thought. I would separate the endpoints and controllers to handle something like:

```
GET /me #show
POST /me # create
PUT /me # update
```
Notice that I made that into a singular resource so you would not pass an ID when updating or showing it. The endpoints can also be something different like `/user` or `/account`. There should be a controller that handles just this.

The next portion would be visiting other user's profiles. There you might need:

```
GET /profiles #index
GET /profiles/:id # show
```

This controller should not allow you to update, delete or create another user's profile. A new user wanting to create a profile (sign up) would go through the first controller to do so.

Finally, we can have an admin controller that can do everything you could need to manage users. You could name it after `profiles` or `accounts` or `users` just prepend `admin` in front of it.

```
GET /admin/users # index
GET /admin/users/:id # show
POST /admin/users # create
PATCH /admin/users/:id #update
DELETE /admin/users/:id #destroy
```